src/Jackett.Common/Utils/StringUtil.cs

```diff
@@ -138,13 +138,35 @@ public static string MakeValidFileName(string text, char? replacement = '_', boo
             return changed ? sb.ToString() : text;
         }
 
-        public static string GetQueryString(this NameValueCollection collection, Encoding encoding = null) =>
-            string.Join("&", collection.AllKeys.Select(a =>
-                $"{a}={WebUtilityHelpers.UrlEncode(collection[a], encoding ?? Encoding.UTF8)}"));
+        /// <summary>
+        /// Converts a NameValueCollection to an appropriately formatted query string.
+        /// Duplicate keys are allowed in a NameValueCollection, but are stored as a csv string in Value.
+        /// This function handles leaving the values together in the csv string or splitting the value into separate keys
+        /// </summary>
+        /// <param name="collection">The NameValueCollection being converted</param>
+        /// <param name="encoding">The Encoding to use in url encoding Value</param>
+        /// <param name="splitMultiValues">Whether the Value should be Key=Value or split into Key=Val1&Key=Val2</param>
+        /// <param name="separator">The string used to separate each query value</param>
+        /// <returns>A web encoded string of key=value parameters separated by the separator</returns>
+        public static string GetQueryString(this NameValueCollection collection, Encoding encoding = null,
```

Comment:
Encoding encoding = Encoding.UTF8
and remove `?? Encoding.UTF8` below

Reply:
We should either keep `?? Encoding.UTF8` or change it to `?? throw new ArgumentNullException(nameof(encoding) + " cannot be null")` in case of someone passing in null explicitly. But we can change the default value so it shows up in the tooltip for the function.

Reply:
I guess `WebUtilityHelpers.UrlEncode(` will throw the exception for us if it's not valid encoding or null

Reply:
Actually, this is null because we can't use `Encoding.UTF8` as a default value. Compilation error on the change says `Default parameter must be a compile time constant` which `Encoding.UTF8` is not.

Reply:
oki